packages/history-service/src/base-history.ts

```diff
@@ -148,4 +159,32 @@ export abstract class BaseHistory implements ConsumerHistory {
       rootHistoryKey: this.rootHistory.location.key
     };
   }
+
+  private updateRootHistoryKey(oldKey: string): void {
```

Comment:
You can use `this.currentRootHistoryKey` instead of `oldKey` argument

Reply:
I had to think long and hard about all these auxiliary functions. I don't find them very understandable. For example: The same root location is accessed once via instance variable and once via location change handler. In addition, multiple locations are updated and not just one. My requested changes would have made my initial understanding easier.

Reply:
I partially agree that this is not optimal yet. Why do you say that multiple locations are updated and not just one?

Reply:
Also, I wanted to contain the handling of `currentRootHistoryKey` in the `handleRootLocationChange` method, i.e. checking for `undefined` and updating it. `updateRootHistoryKey` on the other hand is only responsible for calling `setRootHistoryKey` for the location that matches the `oldKey`.